### PR TITLE
Add unit tests for parser

### PR DESCRIPTION
**Pull request by quangkhoa**

### CODE REVIEW

File: src/parser/open-api-v3/open-api-v3-parser.js [hunk 1]
```diff
@@ -2,54 +2,113 @@ import refParser from 'json-schema-ref-parser';
...
+ * Sort function
+ *
+ * @param {String} inMethod1
+ * @param {String} inMethod2
+ *
+ * @return {number}
+ */
+
+function defaultMethodSort(inMethod1, inMethod2) {
+  const methodWeights = {
```
Comment (brendo) [inline src/parser/open-api-v3/open-api-v3-parser.js]: `methodWeights` can be extracted out as a constant for the project. Right now the variable will be created each time this function is called. Minor perf thing.

File: src/parser/open-api-v3/open-api-v3-parser.js [hunk 2]
```diff
@@ -2,54 +2,113 @@ import refParser from 'json-schema-ref-parser';
...
+    HEAD: 5,
+    OPTIONS: 6,
+    TRACE: 7,
+    CONNECT: 8
+  };
+
+  const method1 = inMethod1.toUpperCase();
+  const method2 = inMethod2.toUpperCase();
+
+  if (!methodWeights.hasOwnProperty(method1)) {
```
Comment (brendo) [inline src/parser/open-api-v3/open-api-v3-parser.js]: Not sure the sort function is the appropriate place for filtering out bad methods?

**Review by brendo — CHANGES_REQUESTED** [review]
Consider using [swagger2openapi](https://github.com/mermade/swagger2openapi) to generate an additional fixture for Petstore to ensure we have better coverage for parts of the spec we may not use internally.

File: src/parser/open-api-v3/open-api-v3-parser.js (hunk 2 above)
Comment (quangkhoa) [reply src/parser/open-api-v3/open-api-v3-parser.js]: The check to filter out bad methods needs to be done somewhere (to be added). This extra check is only for the sort function to protect itself, i.e. make sure it only deals with valid data.

File: src/parser/open-api-v3/open-api-v3-parser.js (hunk 1 above)
Comment (quangkhoa) [reply src/parser/open-api-v3/open-api-v3-parser.js]: Good pickup. Fixed.

**Comment by quangkhoa** [comment]
@brendo I've updated the test to load input/output data files. This will allow adding more input/output data files in the future. At the moment, our renderer does not support `Array` in schema, and thus will not work well with Petstore. This will be address in https://github.com/temando/open-api-renderer/issues/22. When that issue is resolved, I'll add more  tests to it, including the Petstore example.

File: src/parser/open-api-v3/open-api-v3-parser.js (hunk 2 above)
Comment (quangkhoa) [reply src/parser/open-api-v3/open-api-v3-parser.js]: Removed as requested

File: test/parser/open-api-v3/open-api-v3-parser.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { readdirSync, readJSONSync } from 'fs-extra';
+import { expect } from 'chai';
+// import inputShippingDefinition from './data/definition/inputs/shipping.json';
+// import expectedOutputShippingDefinition from './data/definition/outputs/shipping.json';
+import getUIReadyDefinition from '../../../src/parser/open-api-v3/open-api-v3-parser';
+
+describe.only('getUIReadyDefinition', () => {
```
Comment (brendo) [inline test/parser/open-api-v3/open-api-v3-parser.test.js]: Just need to drop this `only` and you're good to merge 👍

File: test/parser/open-api-v3/open-api-v3-parser.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { readdirSync, readJSONSync } from 'fs-extra';
+import { expect } from 'chai';
+// import inputShippingDefinition from './data/definition/inputs/shipping.json';
```
Comment (brendo) [inline test/parser/open-api-v3/open-api-v3-parser.test.js]: Clean up these unused imports too.

File: test/parser/open-api-v3/open-api-v3-parser.test.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+describe.only('getUIReadyDefinition', () => {
+  const dataDirectory = __dirname + '/data/definition';
+  const dataFiles = readdirSync(dataDirectory + '/inputs');
+
+  const tests = [];
+
+  dataFiles.map(dataFile => {
+    try {
+      const inputData = readJSONSync(dataDirectory + '/inputs/' + dataFile);
+      const outputData = readJSONSync(dataDirectory + '/outputs/' + dataFile);
```
Comment (brendo) [inline test/parser/open-api-v3/open-api-v3-parser.test.js:14]: Nice 👍

File: test/parser/open-api-v3/open-api-v3-parser.test.js (hunk 3 above)
Comment (quangkhoa) [reply test/parser/open-api-v3/open-api-v3-parser.test.js]: Dropped

File: test/parser/open-api-v3/open-api-v3-parser.test.js (hunk 4 above)
Comment (quangkhoa) [reply test/parser/open-api-v3/open-api-v3-parser.test.js]: Removed

File: test/parser/open-api-v3/open-api-v3-parser.test.js (hunk 5 above)
Comment (quangkhoa) [reply test/parser/open-api-v3/open-api-v3-parser.test.js:14]: Thank you

**Pull request merged**